lib/ansible/plugins/test/core.py

```diff
@@ -149,7 +149,9 @@ def regex(value='', pattern='', ignorecase=False, multiline=False, match_type='s
     if multiline:
         flags |= re.M
     _re = re.compile(pattern, flags=flags)
-    return bool(getattr(_re, match_type, 'search')(value))
+    if not callable(method := getattr(_re, match_type, None)):
```

Comment:
i would switch this to `try`/`except AttributeError` , just to optimize the happy path